oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Commit.java

```diff
@@ -354,7 +357,42 @@ private void applyToDocumentStore(RevisionVector baseBranchRevision)
         boolean commitRootHasChanges = operations.containsKey(commitRootPath);
         for (UpdateOp op : operations.values()) {
             NodeDocument.setCommitRoot(op, revision, commitRootDepth);
+
+            // special case for :childOrder updates
+            if (store.isCommitCleanupFeatureEnabled()) {
+                final Branch localBranch = getBranch();
+                if (localBranch != null) {
+                    final TreeSet<Revision> commits = new TreeSet<>(localBranch.getCommits());
```

Comment:
```suggestion
                    final NavigableSet<Revision> commits = new TreeSet<>(localBranch.getCommits());
```

Reply:
I believe the feature toggle needs to be on the DocumentNodeStore, not the DocumentStore.